lib/CachedSource.js

```diff
@@ -30,8 +30,10 @@ class CachedSource extends Source {
 
 	size() {
 		if(typeof this._cachedSize !== "undefined") return this._cachedSize;
-		if(typeof this._cachedSource !== "undefined")
-			return this._cachedSize = this._cachedSource.length;
+		if(typeof this._cachedSource !== "undefined") {
+			if(Buffer.from.length === 1) return new Buffer(this._cachedSource).length;
```

Comment:
```diff
- new Buffer
+ Buffer.from
```

Reply:
Tried that, but that causes the tests to fail for pre node 5.10. `Buffer.from.length` checks if it supports `Buffer.from` and if it is equal to 1, it doesn't

Reply:
And if it is after 5.10, it supports `Buffer.byteLength`.